### PR TITLE
[SPARK-29053][WEBUI][2.4] Sort does not work on some columns

### PR DESCRIPTION
**Pull request by amanomer**
### What changes were proposed in this pull request?
Setting custom sort key for duration and execution time column.

### Why are the changes needed?
Sorting on duration and execution time columns consider time as a string after converting into readable form which is the reason for wrong sort results as mentioned in [SPARK-29053](https://issues.apache.org/jira/browse/SPARK-29053).

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
Test manually.

Back-port of commit 93ac4e1b2df5ca1dde29010320c6faead537a7b1

### CODE REVIEW

**Comment by amanomer** [comment]
cc @srowen

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by wangyum** [comment]
@amanomer Could you rename PR title to `[SPARK-29053][WEBUI][2.4] Sort does not work on some columns`?

**Comment by amanomer** [comment]
I will fix sorting issue in ThriftServerSessionPage as suggested in https://github.com/apache/spark/pull/25892.

**Comment by amanomer** [comment]
Thanks @wangyum

**Comment by wangyum** [comment]
@amanomer Please add `[SPARK-28599]` to PR title.

**Comment by srowen** [comment]
@wangyum does this really address SPARK-28599? this seems to just sort the current page? which brings up an issue of whether that's confusing (or maybe I misunderstood). In any event, this one is just a back-port; does the original PR address it too?

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/ThriftServerSessionPage.scala [hunk 1]
```diff
@@ -100,7 +100,8 @@ private[ui] class ThriftServerSessionPage(parent: ThriftServerTab)
           <td>{info.groupId}</td>
           <td>{formatDate(info.startTimestamp)}</td>
           <td>{formatDate(info.finishTimestamp)}</td>
-          <td>{formatDurationOption(Some(info.totalTime))}</td>
```
Comment (wangyum) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/ThriftServerSessionPage.scala]: @srowen This change to backport [SPARK-28599](https://github.com/apache/spark/pull/25892). 
Or we should revert this change to make another PR to make it clear?

Comment (srowen) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/ThriftServerSessionPage.scala]: I see, these are really duplicates? I would first merge https://github.com/apache/spark/pull/25892 then, but, hm, how is it not a merge conflict? I'm missing something.

My larger point was: does this actually mean you sort all of the entries when there are many pages of them? or just the current page?

Comment (wangyum) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/ThriftServerSessionPage.scala]: https://github.com/apache/spark/pull/25855 for `ThriftServerPage`, https://github.com/apache/spark/pull/25892 for `ThriftServerSessionPage`. We can support all pages in `ThriftServerTab` after this PR.

Comment (srowen) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/ThriftServerSessionPage.scala]: Oh I see. Well, want to at least merge #25892 first and then include it here? we can back-port it separately, too.

Comment (dongjoon-hyun) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/ThriftServerSessionPage.scala]: I'll review #25892 now and merge it soon. And, +1 for backporting separately.

Comment (amanomer) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/ThriftServerSessionPage.scala]: Ok. I have reverted my last commit and created another PR https://github.com/apache/spark/pull/25893 to back-port [SPARK-28588](https://issues.apache.org/jira/browse/SPARK-28599) this to branch-2.4. 
Kindly review it

**Comment by amanomer** [comment]
@srowen @dongjoon-hyun

**Comment by srowen** [comment]
Merged to 2.4

**Comment by amanomer** [comment]
Thank you @srowen

**Comment by dongjoon-hyun** [comment]
Thank you all!
